site/src/components/UserAutocomplete/UserAutocomplete.tsx

```diff
@@ -16,45 +17,99 @@ import {
 import { useQuery } from "react-query";
 import { prepareQuery } from "utils/filters";
 
-export type UserAutocompleteProps = {
-	value: User | null;
-	onChange: (user: User | null) => void;
-	label?: string;
+// The common properties between users and org members that we need.
+export type SelectedUser = {
+	avatar_url: string;
+	email: string;
+	username: string;
+};
+
+export type CommonAutocompleteProps<T extends SelectedUser> = {
 	className?: string;
-	size?: ComponentProps<typeof TextField>["size"];
+	label?: string;
+	onChange: (user: T | null) => void;
 	required?: boolean;
+	size?: ComponentProps<typeof TextField>["size"];
+	value: T | null;
 };
 
+export type UserAutocompleteProps = CommonAutocompleteProps<User>;
+
 export const UserAutocomplete: FC<UserAutocompleteProps> = ({
 	value,
-	onChange,
-	label,
-	className,
-	size = "small",
-	required,
+	...props
 }) => {
-	const [autoComplete, setAutoComplete] = useState<{
-		value: string;
-		open: boolean;
-	}>({
-		value: value?.email ?? "",
-		open: false,
-	});
+	const [filter, setFilter] = useState<string>();
+
 	const usersQuery = useQuery({
 		...users({
-			q: prepareQuery(encodeURI(autoComplete.value)),
+			q: filter !== undefined ? prepareQuery(encodeURI(filter)) : "",
 			limit: 25,
 		}),
-		enabled: autoComplete.open,
+		enabled: filter !== undefined,
 		keepPreviousData: true,
 	});
+	return (
+		<InnerAutocomplete<User>
+			users={usersQuery.data?.users}
+			value={value}
+			setFilter={setFilter}
+			{...props}
+		/>
+	);
+};
+
+export type MemberAutocompleteProps =
+	CommonAutocompleteProps<OrganizationMemberWithUserData> & {
+		organizationId: string;
+	};
+
+export const MemberAutocomplete: FC<MemberAutocompleteProps> = ({
```

Comment:
Is this colocation bad?  I originally was going to create separate directories for the user and member autocompletes but then I was not sure what to do with `InnerAutocomplete` because I do not really want that to be exported.  Maybe this is a case where just duplicating the code is better, but my sense is that we would rather not have drift here between the two components.

Third option is to export `InnerAutocomplete` as `UserAutocomplete` and the caller is responsible for passing in the data instead of providing components that do it.

Reply:
I think it’s fine for now to keep it under UserAutocomplete. You made some good points. 👍